RecoLocalCalo/EcalRecAlgos/src/EcalIntercalibConstantsGPU.cc

```diff
@@ -0,0 +1,40 @@
+#include "RecoLocalCalo/EcalRecAlgos/interface/EcalIntercalibConstantsGPU.h"
+
+#include "FWCore/Utilities/interface/typelookup.h"
+#include "HeterogeneousCore/CUDAUtilities/interface/cudaCheck.h"
+
+EcalIntercalibConstantsGPU::EcalIntercalibConstantsGPU(EcalIntercalibConstants const& values)
+    : valuesEB_{values.barrelItems()}, valuesEE_{values.endcapItems()} {}
```

Comment:
I think these have all been resolved by moving to `ConvertingESProducerT`. @fwyzard ?

Reply:
Ehm, yes, I guess so?

Looking at the HCAL modules, those are making a copy in pinned host memory instead of capturing by reference: it uses more memory, but the copy from CPU to GPU should then be faster.

Given that the copy is usually done only once per job, the impact on the performance should be negligible either way.

Reply:
I mean there does not seem to be a ESTransientHandle in the `ConvertingESProducerT` (despite the header being loaded). So would the capture by reference be OK then?

Which HCAL module do you look at?

Reply:
> I mean there does not seem to be a ESTransientHandle in the `ConvertingESProducerT` (despite the header being loaded). So would the capture by reference be OK then?

I think so, but since @makortel made the original comment, I would leave it to him to confirm :-)

> Which HCAL module do you look at?

All the ones from https://github.com/cms-sw/cmssw/pull/32039, for example
  - [CondFormats/HcalObjects/interface/HcalGainsGPU.h](https://github.com/cms-sw/cmssw/blob/master/CondFormats/HcalObjects/interface/HcalGainsGPU.h)
  - [CondFormats/HcalObjects/src/HcalGainsGPU.cc](https://github.com/cms-sw/cmssw/blob/master/CondFormats/HcalObjects/src/HcalGainsGPU.cc)

Reply:
By the way, those are the same set of changes I was referring to [earlier](https://github.com/cms-sw/cmssw/pull/31719#issuecomment-742571557).

One reason for making the change before the integration would be to avoid creating the files under `RecoLocalCalo/EcalRecAlgos` and soon after moving them to `CondFormats/EcalObjects` where they likely belong.

On the other hand, if this is the only remaining issue, it can indeed be addressed by a follow up PR.

Reply:
Right, using `ConvertingESProducerT` makes the use references safe here.